### ISSUE TITLE
Taxonomy loading incorrect tags

### ISSUE THREAD

**Issue by therevolver** [WordPress/gutenberg#12917]
Years ago I renamed the category taxonomy to something else and since added a custom taxonomy called "categories" that has the title "Type". Works perfectly fine everywhere in wordpress; classic edtitor, posts list, taxonomy pages, except the Gutenberg editor.

For whatever reason Gutenberg lists my "Type" tags box in the editor, with the correct title, however when I use it to add terms frm that taxonomy it only comes up with terms from the original, renamed category taxonomy.

Hopefully this all makes sense, but I can't quite understand how everywhere in wordpress it works perfectly, but in Gutenberg it loads the correct title but incorrect terms.

**Comment by swissspidy** [comment]
Please share the **full code** you are using to register your taxonomy and unregister the default category taxonomy.

Also, double check that `show_in_rest` is true for your custom taxonomy when you register it.

**Comment by therevolver** [comment]
I've added the code used to register the taxonomy below. Also just to be clear I "renamed" the default category taxonomy. Both have worked fine for years and still do, but not in Gutenberg.

Currently, I'm having to go to the all posts page to add tags under that taxonomy. If I attempt to do it on the post page, as mentioned all I get are the category terms, even though I'm entering the information into a taxonomy box called "Types".

I've also attached a screenshot of what I'm getting. Those two marked taxonomies are bringing up the same list of terms.

`add_action( 'init', 'create_extra_taxonomies' );function create_extra_taxonomies() {register_taxonomy('categories','post',array('label' => 'Types','hierarchical' => false,'show_in_rest' => true));}
![scren](https://user-images.githubusercontent.com/25497554/50052740-ce9f1e00-00f6-11e9-92a4-dac7914a3d3a.jpg)
`

**Comment by swissspidy** [comment]
It's not really renaming. You created a completely new taxonomy :-)

The problem here is that the built-in `category` taxonomy uses `categories` as the `rest_base`, which makes them available via `/wp/v2/taxonomies/categories` in the REST API.

At the same time you register a taxonomy called `categories` without a specific `rest_base`. In this case, WordPress just re-uses `categories` as the `rest_base` — as you can see, this conflicts with the original `category` taxonomy.

What you have to do is define a custom `rest_base` for your taxonomy [as explained here](https://developer.wordpress.org/rest-api/extending-the-rest-api/adding-rest-api-support-for-custom-content-types/#registering-a-custom-post-type-with-rest-api-support). You could name it `types` or `my-categories` or whatever. Just pick something that's not already being used.

**Comment by therevolver** [comment]
Okay, I was just confused as it seemed to work fine up until Gutenberg. I'll attempt to define a specific rest base and see what happens.

**Comment by swissspidy** [comment]
Yeah with Gutenberg everything works via the REST API. That wasn't the case before. The new editor just exposes a naming conflict that has been there all the time :-)

**Comment by therevolver** [comment]
So I did add in 'rest_base' => 'types', which I recall doing previously but it does not seem to solve the problem. It is now not loading types terms at all. In previous posts that have terms, they're not being listed, and when I attempt to add them nothing from the "types" taxonomy is coming up to select.

**Comment by swissspidy** [comment]
Ah, I see `types` is also already taken. It lists all existing post types of your site. You can verify this by opening `https://yourwebsite.com/wp-json/wp/v2/types` in your browser. Pick something that doesn't exist yet.
